File: frontend/src/api/ai.ts
import { api, withBranch } from './client'

export interface AiStatus {
  enabled: boolean
}

export interface AiFieldSuggestion {
  field_name: string
  description: string
}

export interface AiDescribeResponse {
  description: string
  field_suggestions: AiFieldSuggestion[]
}

export interface AiAskSource {
  title: string
  entity_type: string
  route_path: string
}

export interface AiAskResponse {
  answer: string
  sources: AiAskSource[]
  semantic_used: boolean
}

export const aiApi = {
  status: (slug: string) =>
    api.get<AiStatus>(`/projects/${slug}/ai/status`),

  describeEvent: (slug: string, eventId: string, branchId?: string | null) =>
    api.post<AiDescribeResponse>(
      withBranch(`/projects/${slug}/ai/describe-event`, branchId),
      { event_id: eventId },
    ),

  describeEventType: (slug: string, eventTypeId: string, branchId?: string | null) =>
    api.post<AiDescribeResponse>(
      withBranch(`/projects/${slug}/ai/describe-event-type`, branchId),
      { event_type_id: eventTypeId },
    ),

  ask: (slug: string, question: string, branchId?: string | null) =>
    api.post<AiAskResponse>(
      withBranch(`/projects/${slug}/ai/ask`, branchId),
      { question },
    ),
}

File: backend/src/tripl/schemas/ai.py
from __future__ import annotations

import uuid

from pydantic import BaseModel, Field


class AiStatusResponse(BaseModel):
    enabled: bool


class AiDescribeEventRequest(BaseModel):
    event_id: uuid.UUID


class AiDescribeEventTypeRequest(BaseModel):
    event_type_id: uuid.UUID


class AiFieldSuggestion(BaseModel):
    field_name: str
    description: str


class AiDescribeResponse(BaseModel):
    description: str
    field_suggestions: list[AiFieldSuggestion]


class AiAskRequest(BaseModel):
    question: str = Field(min_length=3, max_length=500)


class AiAskSource(BaseModel):
    title: str
    entity_type: str
    route_path: str


class AiAskResponse(BaseModel):
    answer: str
    sources: list[AiAskSource]
    semantic_used: bool
